src/main/java/com/sps/app/servlets/AddPoem.java

```diff
@@ -0,0 +1,78 @@
+package com.sps.app.servlets;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.StringValue;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.sps.poem.PoemLine;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Random;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for adding a new poem. */
+@WebServlet("/new-poem")
+public class AddPoem extends HttpServlet {
+  private String ParsePoemLines(final JsonArray lines) {
+    final List<PoemLine> poemLines = new ArrayList<>();
+    for (int i = 0; i < lines.size(); i++) {
+      final long lineId = Math.abs(new Random().nextLong());
+      poemLines.add(new PoemLine(lineId, lines.get(i).getAsString()));
+    }
+    final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+    return gson.toJson(poemLines);
+  }
```

Comment:
I prefer keeping the private methods to the bottom of the file and the public methods towards the top.